src/core/Akka.Discovery/Config/ConfigServiceDiscovery.cs

```diff
@@ -52,35 +54,79 @@ public ConfigServiceDiscovery(ExtendedActorSystem system)
             var servicePath = config.GetString("services-path");
             if (string.IsNullOrWhiteSpace(servicePath))
             {
-                log.Warning(
+                _log.Warning(
                     "The config path [akka.discovery.config] must contain field `service-path` that points to a " +
                     "configuration path that contains an array of node services for Discovery to contact.");
-                _resolvedServices = new Dictionary<string, Resolved>();
+                _resolvedServices = ImmutableDictionary<string, Resolved>.Empty;
             }
             else
             {
                 var services = system.Settings.Config.GetConfig(servicePath);
                 if (services == null)
                 {
-                    log.Warning(
+                    _log.Warning(
                         "You are trying to use config based discovery service and the settings path described in\n" +
                         $"`akka.discovery.config.services-path` does not exists. Make sure that [{servicePath}] path \n" +
                         "exists and to fill this setting with pre-defined node addresses to make sure that a cluster \n" +
                         "can be formed");
-                    _resolvedServices = new Dictionary<string, Resolved>();
+                    _resolvedServices = ImmutableDictionary<string, Resolved>.Empty;
                 }
                 else
                 {
-                    _resolvedServices = ConfigServicesParser.Parse(services);
+                    _resolvedServices = ConfigServicesParser.Parse(services).ToImmutableDictionary();
                     if(_resolvedServices.Count == 0)
-                        log.Warning(
+                        _log.Warning(
                             $"You are trying to use config based discovery service and the settings path [{servicePath}]\n" +
                             "described `akka.discovery.config.services-path` is empty. Make sure to fill this setting \n" +
                             "with pre-defined node addresses to make sure that a cluster can be formed.");
                 }
             }
 
-            log.Debug($"Config discovery serving: {string.Join(", ", _resolvedServices.Values)}");
+            _log.Debug($"Config discovery serving: {string.Join(", ", _resolvedServices.Values)}");
+        }
+
+        public bool TryRemoveEndpoint(string serviceName, ResolvedTarget target)
```

Comment:
Added this so we can add and remove new discovery entry during runtime, used for MNTR testing.